src/internal.rs

```diff
@@ -37,6 +41,58 @@ impl Default for FileTypes {
     }
 }
 
+enum SizeLimitType {
+    Max,
+    Min,
+}
+
+pub struct SizeFilter {
+    size: u64,
+    limit_type: SizeLimitType,
+}
+
+impl SizeFilter {
+    pub fn is_within(&self, size: u64) -> bool {
+        match self.limit_type {
+            SizeLimitType::Max => size <= self.size,
+            SizeLimitType::Min => size >= self.size,
+        }
+    }
+}
+
+const KILO: u64 = 1024;
+const MEGA: u64 = KILO * 1024;
+const GIGA: u64 = MEGA * 1024;
+const TERA: u64 = GIGA * 1024;
+
+impl<'a> From<&'a str> for SizeFilter {
+    /// Create the `SizeFilter` from the given `&str`.
+    /// It is imperative that the incoming value has been validated for
```

Comment:
Could we move the verification of the format here in this function (`from`) and return `Option<Self>`?

Reply:
Well, we could verify the pattern and call `error` if it was not correct, since it is something we fail for anyway. According to the docs `From<T>` must not fail. I could switch out the implementation for something returning `Option<Self>` also, but it comes down to whether we want to fail & report for incorrect parameters or continue silently

Reply:
Maybe we just provide a `SizeFilter::from_string` method instead of implementing `From`? This method could then return `Option<Self>` and we could move the error handling / error output to the call site (the main function).

Reply:
Sure, sounds good